pinot-core/src/main/java/org/apache/pinot/core/common/datatable/BaseDataTable.java

```diff
@@ -44,21 +44,24 @@
   protected DataSchema _dataSchema;
   protected int[] _columnOffsets;
   protected int _rowSizeInBytes;
+  // _dictionaryMap is only used in V2/V3, where each column use one dictionary to map String->Integer
   protected Map<String, Map<Integer, String>> _dictionaryMap;
+  // _dictionary is only used in V4, where all columns use a common dictionary to map String->Integer
+  protected Map<Integer, String> _dictionary;
```

Comment:
We can further optimize this into a `List`

Reply:
That's a good idea, using the index of list as an implicit dicId.